### PR TITLE
Improve handling of empty line data.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR follows #1041

### CODE REVIEW

**Comment by geoffthemedio** [comment]
Builds and seems to run OK.

**Comment by LGM-Doyle** [comment]
Rebased.

File: GG/GG/Edit.h [hunk 1]
```diff
@@ -179,6 +179,10 @@ class GG_API Edit : public TextControl
     /** Returns the cursor position at the time of the most recent
         double-button-down. */
     std::pair<CPSize, CPSize> DoubleButtonDownCursorPos() const;
+
+    /** Return the index of the last LineData() or 0 if LineData is empty.
+     Allows index based Edit to handle empty line data.*/
+    std::vector<GG::Font::LineData>::size_type LastLineIndexOrZero() const;
```
Comment (geoffthemedio) [inline GG/GG/Edit.h]: Could be called NumLines() perhaps?

File: GG/src/Edit.cpp [hunk 2]
```diff
@@ -338,6 +336,12 @@ bool Edit::InDoubleButtonDownMode() const
 std::pair<CPSize, CPSize> Edit::DoubleButtonDownCursorPos() const
 { return m_double_click_cursor_pos; }
 
+/** Return the index of the last LineData() or 0 if LineData is empty.
```
Comment (geoffthemedio) [inline GG/src/Edit.cpp]: don't need to repeat the comment before the implementation...

File: GG/src/MultiEdit.cpp [hunk 3]
```diff
@@ -453,6 +456,10 @@ Y MultiEdit::BottomMargin() const
 
 std::pair<std::size_t, CPSize> MultiEdit::CharAt(const Pt& pt) const
 {
+
```
Comment (geoffthemedio) [inline GG/src/MultiEdit.cpp]: remove blank line at start

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

File: GG/GG/Edit.h (hunk 1 above)
Comment (LGM-Doyle) [reply GG/GG/Edit.h]: Certainly, `NumLines()` is the more obvious name.

**Comment by LGM-Doyle** [comment]
The requested changes have been made.  The next PR in the series is #989,

**Pull request merged**